### ISSUE TITLE
opt-related detritus

### ISSUE THREAD

**Issue by chdiza** [Homebrew/brew#2427]
- [x] Ran `brew update` and retried your prior step?
- [x] Ran `brew doctor`, fixed as many issues as possible and retried your prior step?
- [x] Confirmed this is a problem with Homebrew/brew and not specific formulae? If it's a formulae-specific problem please file this issue at https://github.com/Homebrew/homebrew-core/issues/new

If I do `brew install jpeg`, and then do `brew rm jpeg`, then there shouldn't be anything from that package left, and in particular, there should be no symlinks in `opt` that point to the `jpeg` keg.

But there are, because if I follow my `brew rm jpeg` with a `brew prune`, the latter tells me that it removed the following dead symlinks:
```
rm /usr/local/opt/libjpeg
rm /usr/local/opt/libjpg
Pruned 2 symbolic links from /usr/local
```
Those are (I guess) aliases for `jpeg`; they point at the same keg that `jpeg` points at.

Long story short: if aliases to a formula `foo` are going to appear in `opt` alongside the canonically-named symlink to `foo`, then `brew remove` should remove them too.

Also, there is in `opt` a symlink `jpeg@8` that *also* links to `jpeg`.  That is weird, since I didn't install a versioned formula.  I didn't `brew install jpeg@8`, I did `brew install jpeg`.  It is also the only symlink in my `opt` that contains an "@", even though I have 59 other packages installed.

Note that this `jpeg@8` symlink *does* get removed by `brew remove`, even though the two mentioned above don't.

**Comment by chdiza** [comment]
(Also, there is no such formula as `jpeg@8`.)

**Comment by ilovezfs** [comment]
@chdiza thanks for reporting this. The jpeg@8 is just a convenience versioned alias for the unversioned jpeg formula. You are correct that all three of the aliases should be cleaned up on brew uninstall, not just one of them with two left behind.

**Comment by MikeMcQuaid** [comment]
Thanks for the great report @chdiza. Fix coming right up.
